tests/cpp/cache_eviction.cpp

```diff
@@ -227,11 +279,28 @@ TEST_P(CacheEvictionLowScoreBlocksParameterizedTest, EvictsLowestScoredBlocks) {
     auto test_evicted_blocks = algo.evict_logical_blocks();
     auto ref_evicted_blocks = test_struct.ref_evicted_blocks;
     for (size_t layer_idx = 0; layer_idx < num_decoder_layers; layer_idx++) {
-        EXPECT_EQ(test_evicted_blocks[layer_idx], ref_evicted_blocks[layer_idx]);
+        // if kvcrush is enabled
+        bool should_apply_kvcrush =
+            (test_struct.eviction_config.get_kvcrush_budget() > 0) &&
+            (std::ceil(static_cast<double>(test_struct.tokens_over_max_cache_size) / DEFAULT_BLOCK_SIZE) >=
+             (test_struct.eviction_config.get_kvcrush_budget()));
+        if (should_apply_kvcrush) {
+            // check test_evicted_blocks is a subset of ref_evicted_blocks
+            EXPECT_TRUE(std::includes(ref_evicted_blocks[layer_idx].begin(),
+                                      ref_evicted_blocks[layer_idx].end(),
+                                      test_evicted_blocks[layer_idx].begin(),
+                                      test_evicted_blocks[layer_idx].end()));
+            EXPECT_EQ(static_cast<int>(test_evicted_blocks[layer_idx].size()),
+                      static_cast<int>(ref_evicted_blocks[layer_idx].size()) -
+                          static_cast<int>(test_struct.eviction_config.get_kvcrush_budget()));
```

Comment:
The problem with these test requirements remains that these are potentially flaky (run-to-run unstable) because of the RNG involved. However, now that you have extracted the RNG controls and able to set a specific constant seed for the sake of testing - instead of these `EXPECT_` conditions, compare exact evicted block indices like it is done in the rest of the cases in this test. To learn which blocks get evicted, please run your new KVCrush test cases in the debugger or use print statements - you will also have an opportunity to mentally check if the outputs are correct before commiting them as references to the test cases.